include/swift/AST/Decl.h

```diff
@@ -3560,6 +3560,9 @@ class GenericTypeParamDecl final
     return *getTrailingObjects<TypeRepr *>();
   }
 
+  /// A name for this generic parameter, accounting for opaqueness.
+  std::string getUserFacingName() const;
```

Comment:
Instead of adding this, just pass the GenericTypeParamType to the diagnostic instead

Reply:
Thanks. this cleaned things up a bunch.